vendor/github.com/aws/aws-sdk-go-v2/aws/awserr/error.go

```diff
@@ -1,145 +0,0 @@
-// Package awserr represents API error interface accessors for the SDK.
```

Comment:
Ideally a library should not vendor its dependencies: that's the role of applications, particularly productionized applications. While this project provides an awsenv binary, it's not, afaik, used in production in any critical way.

Reply:
Actually, I take that back. This is a public repo, and so we don't know how this is being used. Also, every time we need to break compatibility by creating a v2, we should consider that a costly investment, and so we should review the API to see there are aspects to improve, again, particularly if this is a public library.

This is similar to paying hundreds of millions of dollars to give your house a gold plated exterior, while neglecting to address any structural issues it may have: if you're going to ask the users of this library to invest time in switching their imports (or may ask new users who start with v2 to change to a cleaned up v3 later), you might as well take the opportunity to combine quality improvements all at once into a single breaking release.

Reply:
Gotcha, so it sounds like you're suggesting that if we need to make a breaking change we should fix other issues in this library while we're at it? Is that an accurate summary? If so, are there any suggestions on things we should fix? 

From my perspective we're using the latest aws-sdk-go V2 in one of our projects and I needed to update this library to be compatible. So we're currently pointed to this branch and I'd like to get this merged in and get us pointed to the main branch if feasible...before I forget about this 😄